torch/_inductor/codegen/wrapper.py

```diff
@@ -1338,6 +1341,12 @@ def make_buffer_allocation(self, buffer):
             f"{size}, {stride}, at::TensorOptions({device}).dtype({dtype}));"
         )
 
+    def codegen_multi_output(self, name, value):
+        # MultiOutput is allocated by a fallback op and needs to be copied into the final output in the AOT mode
+        if V.graph.aot_mode and name in set(V.graph.get_output_names()):
+            self.outputs_need_copy.add(name)
```

Comment:
I'm not main reviewer for this (unless you want me to be), but conditioning on `aot_mode` here doesn't feel very good to me. It seems like the meaning of `get_output_names()` subtly changes between aot mode and not, and it's likely to be a bug farm unless you have some way of mechanically letting people know how to do things correctly.

Given our recent discussions about ownership, is it really that necessary to prevent model.so from allocating outputs? It feels like this shouldn't be too hard to do.